components/file_search_results/file_search_result_item.tsx

```diff
@@ -0,0 +1,168 @@
+    }
+
+    private jumpToConv = (e: MouseEvent) => {
+        e.stopPropagation();
+        browserHistory.push(`/${this.props.teamName}/pl/${this.props.fileInfo.post_id}`);
+    }
+
+    private copyLink = () => {
+        copyToClipboard(`${getSiteURL()}/${this.props.teamName}/pl/${this.props.fileInfo.post_id}`);
+    }
+
+    private stopPropagation = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.stopPropagation();
+    }
+
+    private keepOpen = (open: boolean) => {
+        this.setState({keepOpen: open});
+    }
+
+    public render(): React.ReactNode {
+        const {fileInfo, channelDisplayName, channelType} = this.props;
+        let channelName: React.ReactNode = channelDisplayName;
+        if (channelType === Constants.DM_CHANNEL) {
+            channelName = (
+                <FormattedMessage
+                    id='search_item.file_badge.direct_message'
+                    defaultMessage='Direct Message'
+                />
+            );
+        } else if (channelType === Constants.GM_CHANNEL) {
+            channelName = (
+                <FormattedMessage
+                    id='search_item.file_badge.group_message'
+                    defaultMessage='Group Message'
+                />
+            );
+        }
+
+        return (
+            <div
+                data-testid='search-item-container'
+                className='search-item__container'
+            >
+                <div
+                    className={'FileSearchResultItem' + (this.state.keepOpen ? ' keep-open' : '')}
+                    onClick={() => this.setState({showPreview: true})}
+                >
+                    <FileThumbnail fileInfo={fileInfo}/>
+                    <div className='fileData'>
+                        <div className='fileDataName'>{fileInfo.name}</div>
+                        <div className='fileMetadata'>
+                            {channelName && <Badge className='file-search-channel-name'>{channelName}</Badge>}
+                            <span>{fileSizeToString(fileInfo.size)}</span>
+                            <span>{' • '}</span>
+                            <Timestamp
+                                value={fileInfo.create_at}
+                                ranges={FILE_TOOLTIP_RANGES}
+                            />
+                        </div>
+                    </div>
+                    <OverlayTrigger
+                        delayShow={1000}
+                        placement='top'
+                        overlay={
+                            <Tooltip id='file-name__tooltip'>
+                                {localizeMessage('file_search_result_item.more_actions', 'More Actions')}
+                            </Tooltip>
+                        }
+                    >
+                        <div onClick={this.stopPropagation}>
+                            <MenuWrapper
+                                onToggle={this.keepOpen}
+                            >
```

Comment:
Could be simplified, `MenuWrapper` has a built-in `stopPropagationOnToggle` (minor; maybe for later)

Reply:
Done